crates/slumber_core/src/template/prompt.rs

```diff
@@ -33,6 +35,25 @@ pub struct Prompt {
     /// How the prompter will pass the answer back
     pub channel: PromptChannel<String>,
 }
+/// When the app goes to collect user input on a select list
+/// the expected result is the chosen index from the options vector
+/// HOWEVER, that makes it very tough to render a preview in the TUI
+/// So, we use the below enum to signal that the result of the selection could be either
+#[derive(Debug)]
+pub enum SelectResult {
+    Placeholder(String),
+    Index(usize)
+}
+
+#[derive(Debug)]
+pub struct Select {
+    /// Tell the user what we're asking for
+    pub message: String,
+    /// List of choices the user can pick from
+    pub options: Vec<String>,
+    /// How the prompter will pass the answer back
+    pub channel: PromptChannel<SelectResult>,
```

Comment:
I'm thinking it might make more sense to just passing a `String` back, and ditch `SelectResult`. I see two benefits for this:
- Makes it simpler, eliminates the need for `SelectResult`
- Eliminates the clones in `render_select`. The creator of this passes the owned strings as part of `options`, and the implementor would pass back one of those owned strings

Reply:
ah, yea - I wrestled with that idea for a bit. Happy to implement as just a string! I didn't love building the enum, but wasn't sure how important maintaining the index-return signature might be. Allowing the implementer to just return the actual chosen string instead of the index makes good sense!

Reply:
Interestingly (and probably not a huge deal) but allowing a `String` response over the Select channel means the implementor could choose any arbitrary string even one potentially outside any of the original `options` 🤷 

It'd be fairly easy (famous last words) to re-validate that the response string is IN the original list of options during the `render_select` function after it receives the response over the channel, but I don't feel particularly strong that such validation is needed right now.

Reply:
Agreed. The lack of constraints on the type side does open the possibility of incorrect behavior, but also we need to support placeholder strings _somehow_, and for 100% correctness we would have to validate the index anyway. Since this is an internal API I think just taking a `String` is fine.